src/streams/send.rs

```diff
@@ -696,6 +703,28 @@ impl StreamTx {
 
         let mut snapshot = self.send_queue.snapshot(now);
 
+        // Outstanding resends
+        let mut resend_snapshot =
+            self.resends
+                .iter()
+                .fold(QueueSnapshot::default(), |mut snapshot, r| {
```

Comment:
Doing this every time doesn't feel great for performance but neither does exposing `SendQueue::total` so we can call `increase` and `decrease` as the resends change.

We could also move `TotalQueue` out of `SendQueue` to `StreamTx`